### ISSUE TITLE
Parse await expression in VB when method is not async

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#10320]
If you use an await expression in VB when the enclosing method is not async, you get an error that you cannot use await in a non-async method, and you **also** get a syntax error. In most cases we could drop the syntax error and just report the former error message.

We could also check that we bind properly if we relax the parser. See C# issue #10232.

**Comment by AnthonyDGreen** [comment]
Agreed. The only ambiguous case is `Await` _ParenthesizedExpression_ because that could hypothetically be an invocation of a method named "Await". In any other case we should just parse it and report the error about the missing Async modifier.

**Comment by markhurd** [comment]
And note #7593 where I report and suggest a fix for `[Await](i)` displaying the wrong error now.
